conduit/pipeline.go

```diff
@@ -228,10 +226,27 @@ func (p *pipelineImpl) Start() error {
 		p.logger.Infof("Initialized Processor: %s", processorName)
 	}
 
-	return p.RunPipeline()
+	go p.runPipeline()
+	return err
 }
 
-func (p *pipelineImpl) Stop() error {
+func (p *pipelineImpl) Stop() {
+	p.cf()
```

Comment:
How is this cancel function set?

Reply:
In `MakePipeline` which takes a context, we create a cancel context and assign it to the pipelineImpl. 

The one thing which I haven't included is an easy way to reset that context. So if you call `Start` and then `Stop` things wouldn't really work if you called `Start` again.

Reply:
I think we just say that once you call stop you are closing conduit.